arduino/main/Arduino to RPi (PUSH).ino

```diff
@@ -0,0 +1,398 @@
+        strcat(IFramesBuffer[index], ",");
+        memset(GyYChar, NULL, 5);
+        dtostrf(GyZ[sensor_loops], 0, 0, GyZChar);      // convert double to char[] with no dp, effectively making it int.
+        strcat(IFramesBuffer[index], GyZChar);
+        strcat(IFramesBuffer[index], ",");
+        memset(GyZChar, NULL, 5);
+      }
+      dtostrf(voltVal, 0, 2, voltChar);      // convert double to char[] with 2dp.
+      strcat(IFramesBuffer[index], voltChar);
+      strcat(IFramesBuffer[index], ",");
+      dtostrf(currentVal, 0, 2, currentChar);// convert double to char[] with 2dp.
+      strcat(IFramesBuffer[index], currentChar);
+      strcat(IFramesBuffer[index], ",");
+      dtostrf(powerVal, 0, 2, powerChar);    // convert double to char[] with 2dp.
+      strcat(IFramesBuffer[index], powerChar);
+      strcat(IFramesBuffer[index], ",");
+      dtostrf(energyVal, 0, 2, energyChar);  // convert double to char[] with 2dp.
+      strcat(IFramesBuffer[index], energyChar);
+      //Serial.print("Current IFrame before checksum is "); Serial.println(IFramesBuffer[index]);
+      byte len = strlen(IFramesBuffer[index]);
+      IFramesBuffer[index][1] &= 0xFE;  // Converts bit[0] back to the original intended 0 for checksum calculation.
+      int checksum = crc16(&IFramesBuffer[index][0], len);
+      IFramesBuffer[index][1] |= 0b1;   // Converts bit[0] to 1 again for complete string storage purposes.
+      IFramesBuffer[index][len++] = checksum >> 8;
+      IFramesBuffer[index][len++] = checksum & 0xFF;
+      IFramesBuffer[index][len] = '\0';
+      Serial.print("Length of I-Frame is "); Serial.println(len);
+      //      bool deleteE = (IFramesBuffer[index][len] != 'E');
+      //      Serial.print("IFramesBuffer[index][len] is: "); Serial.println(IFramesBuffer[index][len]);
+      //      len = strlen(IFramesBuffer[index]);
+      //      Serial.print("Last IFrame character is: "); Serial.println(IFramesBuffer[index][len - 1]);
+      //      if ( (IFramesBuffer[index][len - 1] == 'E') && deleteE) IFramesBuffer[index][len - 1] = '\0'; // Destroy the elusive 'E' character that appears sometimes. Why does this even happen??
+      //      Serial.print("Complete IFrame #"); Serial.print(frameNum); Serial.print(" is "); Serial.println(IFramesBuffer[index]);
+      Serial.print("IFramesBuffer[frameNum] is: "); Serial.write(IFramesBuffer[frameNum]);
+      frameNum = (frameNum + 1) & 0x7F; // keeps the range of frameNum between 0 - 127
+      index = (index + 1) & 0xF;
+      delay(100);
+      xSemaphoreGive(UninterruptedReadSemaphore);
+      vTaskDelayUntil(&prevWakeTimeRead, READ_FREQUENCY);
+    }
+  }
+}
+
+// Reads the accelerometer and gyroscope values from the Queue
+// Checks for an S-Frame so that Arduino knows that it is ok to send data
+void SendValues(void *pvParameters) {
+  byte buf[50];               // buf[0] = START, buf[1] = numReceive, buf[2] = frame, buf[3] = checkNum, buf[4] = checkNum, buf[5] = STOP
```

Comment:
This should send values as soon as they are ready, not when s-frame rr is received. 
Possible implementation: put messages from readSensor into a queue. SendValues should block indefinitely on this queue. When there is a message in the queue it will send data